platform/view/core/sig/deserializer.go

```diff
@@ -32,12 +34,22 @@ func NewMultiplexDeserializer(sp driver.ServiceProvider) (*deserializer, error)
 }
 
 func (d *deserializer) AddDeserializer(newD Deserializer) {
+	d.deserializersMutex.Lock()
 	d.deserializers = append(d.deserializers, newD)
+	d.deserializersMutex.Unlock()
 }
 
 func (d *deserializer) DeserializeVerifier(raw []byte) (driver.Verifier, error) {
 	var errs []error
+
```

Comment:
why don't we use a read lock here? I expect this function to be called many times. A read lock would be faster given that `AddDeserializer` in comparison is called fewer times.

Reply:
👍